sycl/plugins/cuda/pi_cuda.cpp

```diff
@@ -2096,6 +2096,8 @@ pi_result cuda_piextContextCreateWithNativeHandle(pi_native_handle nativeHandle,
   *piContext =
       new _pi_context{_pi_context::kind::user_defined, newContext, device};
 
+  cuda_piContextRetain(*piContext);
```

Comment:
Is this safe? Do we do the same for queues? I think the better solution is to make sure `_pi_context` knows that it doesn't have ownership, because otherwise you may end up leaking the `_pi_context` object, even if the `CUcontext` gets deleted by user-code.

Also, does the backend spec specify who has ownership? I know L0 includes ownership as an option in the input type.

Reply:
Yes you are correct. I have added an ownership variable to `_pi_context` to prevent it from destroying types created with from interop handles. The backend spec says that in these cases it does not own the object.

Reply:
Done